pkg/reconcilermanager/controllers/rootsync_controller.go

```diff
@@ -401,23 +400,13 @@ func (r *RootSyncReconciler) deleteManagedObjects(ctx context.Context, reconcile
 	return nil
 }
 
-// SetupWithManager registers RootSync controller with reconciler-manager.
-func (r *RootSyncReconciler) SetupWithManager(mgr controllerruntime.Manager, watchFleetMembership bool) error {
-	// Index the `gitSecretRefName` field, so that we will be able to lookup RootSync be a referenced `SecretRef` name.
-	if err := mgr.GetFieldIndexer().IndexField(context.Background(), &v1beta1.RootSync{}, gitSecretRefField, func(rawObj client.Object) []string {
-		rs, ok := rawObj.(*v1beta1.RootSync)
-		if !ok {
-			// Only add index for RootSync
-			return nil
-		}
-		if rs.Spec.Git == nil || v1beta1.GetSecretName(rs.Spec.Git.SecretRef) == "" {
-			return nil
-		}
-		return []string{rs.Spec.Git.SecretRef.Name}
-	}); err != nil {
-		return err
-	}
+// Setup RootSync controller with reconciler-manager.
+func (r *RootSyncReconciler) Setup(_ controllerruntime.Manager) error {
+	return nil
```

Comment:
Do you anticipate that `Setup` will need to do something in the future? I'm curious why we need `Setup` as part of the interface if it's not used by RepoSyncReconciler and RootSyncReconciler

Reply:
I pulled it out to put the indexing there, but then discovered indexing didn't work without CRDs either. So I had to remove the indexes completely.

I could change this back to the old SetupWithManager tho, to make the PR a little smaller. I kinda like the explicitness of the interface saying which one happens before and which happens after c-m Start, but it's not required.

Reply:
Removed the Setup method.